src/main/java/org/springframework/data/elasticsearch/core/event/BeforeDeleteCallback.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core.event;
+
+import org.springframework.data.elasticsearch.core.mapping.IndexCoordinates;
+
+/**
+ * Callback that is called before an entity is deleted from the index.
+ *
+ * @author Peter-Josef Meisch
+ * @since 4.0
+ */
+@FunctionalInterface
+public interface BeforeDeleteCallback extends ElasticsearchEntityCallback<String> {
```

Comment:
We typically provide callbacks only for save and delete usecases. In both cases, the callbacks are tied to entities only.